digitalocean/spaces/resource_spaces_key_test.go

```diff
@@ -0,0 +1,169 @@
+				),
+			},
+			{
+				Config: testAccDigitalOceanSpacesKeyConfigWithGrantUpdate(expectedNewName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "name", expectedNewName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.bucket", bucketName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.permission", "read"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDigitalOceanSpacesKey_multipleGrants(t *testing.T) {
+	expectedName := acceptance.RandomTestName()
+	bucketName := acceptance.RandomTestName()
+	bucketName2 := acceptance.RandomTestName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanSpacesKeyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDigitalOceanSpacesKeyConfigMultipleGrants(expectedName, bucketName, bucketName2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "name", expectedName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.bucket", bucketName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.permission", "read"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.1.bucket", bucketName2),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.1.permission", "readwrite"),
+				),
+			},
+			{
+				Config: testAccDigitalOceanSpacesKeyConfig(expectedName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "name", expectedName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.bucket", bucketName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_spaces_key.key", "grant.0.permission", "read"),
+					resource.TestCheckNoResourceAttr(
+						"digitalocean_spaces_key.key", "grant.1.bucket"),
+					resource.TestCheckNoResourceAttr(
+						"digitalocean_spaces_key.key", "grant.1.permission"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDigitalOceanSpacesKeyConfig(name, bucket string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_bucket" "bucket" {
+  name   = "%s"
+  acl    = "public-read"
+  region = "sfo3"
+}
+resource "digitalocean_spaces_key" "key" {
+  name = "%s"
+  grant {
+    bucket     = digitalocean_spaces_bucket.bucket.name
+    permission = "read"
+  }
+}
+`, bucket, name)
+}
+
+func testAccDigitalOceanSpacesKeyConfigWithGrantUpdate(name, bucket string) string {
+	return fmt.Sprintf(`
+resource "digitalocean_spaces_key" "key" {
+  name = "%s"
+  grant {
+    bucket     = "%s"
+    permission = "read"
+  }
+}
+`, name, bucket)
+}
```

Comment:
```suggestion
func testAccDigitalOceanSpacesKeyConfigWithGrantUpdate(bucketName, spacesKeyName, bucketName2 string) string {
	return fmt.Sprintf(`
resource "digitalocean_spaces_bucket" "bucket" {
  name   = "%s"
  acl    = "public-read"
  region = "sfo3"
}

resource "digitalocean_spaces_key" "key" {
  name = "%s"
  grant {
    bucket     = "%s"
    permission = "read"
  }
}
`, bucketName, spacesKeyName, bucketName2)
}
```

Test pass with this^ I _think_ it's failing because leaving out the bucket here will make it seem as tho the bucket doesn't exist. In debug (export TF_LOG=DEBUG) mode I can confirm a PUT request was called:
```

---[ REQUEST ]---------------------------------------
PUT /v2/spaces/keys/xxxx HTTP/1.1
Host: api.digitalocean.com
User-Agent: Terraform/1.5.7 godo/1.138.0
Content-Length: 101
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{
 "name": "tf-acc-test-xxxx",
 "grants": [
  {
   "bucket": "tf-acc-test-xxx",
   "permission": "read"
  }
 ]
}

```

Going to test it some more. It does seem like when a 403 is thrown, it retries indefinitely.